diff -Naur  glib-2.68.1/glib/gspawn.c.org glib-2.68.1/glib/gspawn.c
--- glib-2.68.1/glib/gspawn.c.org	2022-08-19 19:09:04.256844200 +0800
+++ glib-2.68.1/glib/gspawn.c	2022-08-19 19:09:43.469131000 +0800
@@ -1494,7 +1494,7 @@
    *
    * Handle ENOSYS in case it’s supported in libc but not the kernel; if so,
    * fall back to safe_fdwalk(). */
-  if (close_range (lowfd, G_MAXUINT) != 0 && errno == ENOSYS)
+  if (close_range (lowfd, G_MAXUINT, 0) != 0 && errno == ENOSYS)
 #endif  /* HAVE_CLOSE_RANGE */
   (void) safe_fdwalk (close_func, GINT_TO_POINTER (lowfd));
 #endif

